enginetest/queries/order_by_group_by_queries.go

```diff
@@ -320,4 +320,18 @@ var OrderByGroupByScriptTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Group by null = 1",
+		// https://github.com/dolthub/dolt/issues/9035
+		SetUpScript: []string{
+			"create table t0(c0 int, c1 int)",
```

Comment:
Keyless tables can appear in any order.

I would add a test with an `id int primary key column`, so we can more definitively see which rows appear first in the first buffer. 

I'd also add a test with more rows to see how the NULL affects the rest of the rows.